### DISCUSSION TITLE
How do I customize size of SegmentedControl

### DISCUSSION THREAD

**Discussion by mattaningram** [mantinedev/mantine#3882] [Q&A]
I want my `SegmentedControl` components to match sizes with `Input` and `Button` but it appears the height of the control is determined by the height of the `indicator` span inside of it, and that gets calculated automatically and can't be overridden in the usual way in `ThemeProvider` or elsewhere.

I don't understand why the height of the indicator is being calculated and set with JS when it is only sliding horizontally in horizontal mode. It only needs to be set with JS if it is transitioning in vertical mode to an option that has a different height. Even worse the calculations return multi decimal point pixel values (not rem) that don't even match the size of the <label> element it sits on top of. This would be a way simpler component if the size was just done the normal CSS way and then only set and calculated by JS in the specific cases where it is needed (changing size on transition).

Is there no way to have all my controls have equal heights?

**Comment by rtivital** [comment]
> I want my SegmentedControl components to match sizes with Input and Button

You can use Styles API to change internal styles of any Mantine component, including SegmentedControl. You can find an example here – https://codesandbox.io/s/gracious-gwen-1n2bxt?file=/src/App.tsx


> but it appears the height of the control is determined by the height of the indicator span inside of it,

It is a wrong statement. Indicator height is calculated based on label height.


> I don't understand why the height of the indicator is being calculated and set with JS when it is only sliding horizontally in horizontal mode.

You are welcome to explore SegmentedControl to learn how the component is built. It requires js for sizing to calculate elements dimensions as it depends on label font-size font-family and other properties. These values cannot be measured without js.
